crates/ruff_python_stdlib/src/typing.rs

```diff
@@ -321,19 +321,69 @@ type ModuleMember = (&'static str, &'static str);
 /// library (e.g., `list` for `typing.List`), if such a generic was introduced by [PEP 585].
 ///
 /// [PEP 585]: https://peps.python.org/pep-0585/
-pub fn as_pep_585_generic(module: &str, member: &str) -> Option<ModuleMember> {
-    match (module, member) {
-        ("typing", "Dict") => Some(("", "dict")),
-        ("typing", "FrozenSet") => Some(("", "frozenset")),
-        ("typing", "List") => Some(("", "list")),
-        ("typing", "Set") => Some(("", "set")),
-        ("typing", "Tuple") => Some(("", "tuple")),
-        ("typing", "Type") => Some(("", "type")),
-        ("typing_extensions", "Type") => Some(("", "type")),
-        ("typing", "Deque") => Some(("collections", "deque")),
-        ("typing_extensions", "Deque") => Some(("collections", "deque")),
-        ("typing", "DefaultDict") => Some(("collections", "defaultdict")),
-        ("typing_extensions", "DefaultDict") => Some(("collections", "defaultdict")),
+pub fn as_pep_585_generic(call_path: &[&str]) -> Option<ModuleMember> {
+    match call_path {
+        // Builtins
+        ["typing" | "typing_extensions", "Tuple"] => Some(("", "tuple")),
```

Comment:
nit: we can split this into two matches to short-circuit when the call path is not from `typing` / `typing_extensions` module like:
```rs
let ["typing" | "typing_extensions", member @ ..] = call_path else {
    return None;
};

match member {
    // Builtins
    ["Tuple"] => Some(("", "tuple")),
    ["List"] => Some(("", "list")),
	// ...
}
```

Reply:
That's roughly what I did originally [`a818d91` (#5454)](https://github.com/astral-sh/ruff/pull/5454/commits/a818d91fcc6718ec8d4d6c6f5c8c9be136100669#diff-c4f7c61e24d0aea6b6ead5c0c47c17c3a6a126e63ed1bbf3c51b3cc31dee06cdR326-R328)

Not sure if the change back was intentional, or just something lost in the course of the PR...